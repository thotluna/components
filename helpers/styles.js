import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
