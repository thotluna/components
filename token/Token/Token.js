import React from 'react'

import {
  Color,
  Spacing,
  FontFamily,
  FontSize,
  FontWeight,
  LineHeight,
  LetterSpacing,
  BorderRadius,
  BorderWidth,
} from './component'

const Token = () => <></>

Token.Color = Color
Token.Spacing = Spacing
Token.FontFamily = FontFamily
Token.FontSize = FontSize
Token.FontWeight = FontWeight
Token.LineHeight = LineHeight
Token.LetterSpacing = LetterSpacing
Token.BorderRadius = BorderRadius
Token.BorderWidth = BorderWidth

export default Token
