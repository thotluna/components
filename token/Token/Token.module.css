.container {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
}

.description {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-left: 10px;
  gap: 4px;
}

.label {
  min-width: max-content;
  padding: 4px 8px;
  border: 1px solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
  border-radius: var(--border-radius-sm);
  color: var(--color-font-base);
  font-family: var(--font-family-mono);
  font-size: var(--font-size-sm);
}

.label:first-child {
  border: var(--border-width-thin) solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
}

.color {
  width: 65px;
  height: 65px;
  border: var(--button-base-color-border);
  border-radius: var(--border-radius-full);
  box-shadow: var(--box-shadow-sm);
}

.spacer {
  display: flex;
  width: 50px;
  height: 50px;
  align-items: center;
  justify-content: center;
}

.font-size,
.font-family {
  width: 100%;
  max-width: 500px;
  text-align: justify;
}

.font-size {
  text-align: right;
}

.title {
  font-family: var(--font-family-title);
  font-size: 16px;
}

.sans {
  font-family: var(--font-family-sans);
  font-size: 16px;
}

.box {
  width: 20rem;
  height: 5rem;
  border: 1px solid black;
}
