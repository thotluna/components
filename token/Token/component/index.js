export { default as Color } from './Colors/Colors'
export { default as Spacing } from './Spacing/Spacing'
export { default as FontFamily } from './Fonts/FontFamily'
export { default as FontSize } from './Fonts/FontSize'
export { default as FontWeight } from './Fonts/FontWeight'
