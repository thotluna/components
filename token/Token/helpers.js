import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
