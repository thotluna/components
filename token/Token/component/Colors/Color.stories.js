import Token from '../..'

import { choices, decisions } from '../../..'

import { getTemplate, getListTemplate } from '../../../../helpers/storybook'
import { getTokenItems } from '../../helpers'

const Template = getTemplate(Token.Color)
const ListTemplate = getListTemplate(Token.Color)

export default {
  title: 'Tokens/Color',
  component: Token.Color,
  argTypes: {
    value: { control: null },
    label: { control: null },
  },
  parameters: { __sb: { mh: 250 } },
}

const getItems = getTokenItems(choices.color, 'color')

export const Default = Template.bind({})
Default.args = {
  label: '--color-primary',
  value: decisions.color.primary,
}

export const Base = ListTemplate.bind({})
Base.args = { items: getItems('base') }
Base.parameters = {
  __sb: { fd: 'row' },
}

export const Brand = ListTemplate.bind({})
Brand.args = { items: getItems('brand') }
Brand.parameters = {
  __sb: { mh: 300 },
}
