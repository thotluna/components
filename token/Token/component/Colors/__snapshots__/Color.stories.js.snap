// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Tokens/Color Base 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#000",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-base-black
      </span>
      <span
        className="label"
      >
        #000
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-base-white
      </span>
      <span
        className="label"
      >
        #fff
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-base-transparent
      </span>
      <span
        className="label"
      >
        transparent
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Color Brand 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 300,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#B4D1FE",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-blue-angel
      </span>
      <span
        className="label"
      >
        #B4D1FE
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#7B61FF",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-light-slate-blue
      </span>
      <span
        className="label"
      >
        #7B61FF
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#E5E5E5",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-mercury
      </span>
      <span
        className="label"
      >
        #E5E5E5
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#595F68",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-mid-gray
      </span>
      <span
        className="label"
      >
        #595F68
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#676E79",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-mist-blue
      </span>
      <span
        className="label"
      >
        #676E79
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#3B3F45",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-charcoal-gray
      </span>
      <span
        className="label"
      >
        #3B3F45
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#FEB4EF",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-blossom-ping
      </span>
      <span
        className="label"
      >
        #FEB4EF
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#FB0ECC",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-brand-shocking-ping
      </span>
      <span
        className="label"
      >
        #FB0ECC
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/Color Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 250,
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="color"
      style={
        Object {
          "backgroundColor": "#B4D1FE",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --color-primary
      </span>
      <span
        className="label"
      >
        #B4D1FE
      </span>
    </div>
  </div>
</div>
`;
